rfcs/2021-07-19-8216-multiple-pipelines.md

```diff
@@ -257,6 +254,10 @@ Doesn't block to create other sources/sinks.
 
 ## Outstanding Questions
 
+1. Will components defined in the top-level configuration be considered part of a "default" pipeline or no pipeline at all? Sources and sinks seems straightforward to leave out of pipelines, but transforms are a more interesting question. They're the most obvious way to implement filters that limit the data available to specific pipelines.
```

Comment:
By definition, a pipeline shouldn't be able to call another pipeline's transform. So if we put the transforms from the root configuration in a `default` pipeline, it breaks the definition of the pipeline.
To filter we can always do a `where: pipeline_id is null` (or something like this) to get the transforms from the root configuration.
IMO, they should have no pipeline at all.

Reply:
> By definition, a pipeline shouldn't be able to call another pipeline's transform. So if we put the transforms from the root configuration in a `default` pipeline, it breaks the definition of the pipeline.

Yes, this is a very good point. I agree that we should not have a default pipeline, at least from a pure-Vector point of view, but I know that @binarylogic and @zsherman were discussing some other reasons it would be beneficial. We would just need to weigh them against the points here.

Reply:
🤔 didn't have any point about this on the RFC, so maybe this discussion concluded with the same as ours.
I'll write a comment about it on the PR, just in case.